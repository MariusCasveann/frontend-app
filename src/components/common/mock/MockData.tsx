import { GraphQLError } from 'graphql';
import { QueryAllProducts_allProducts } from './../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { QueryAllDomains_allDomains } from './../../dashboard/atoms/DomainSelection/graphql/__generated__/QueryAllDomains';
import { LightApplicationsOfProduct_lightApplicationsOfProduct } from './../../applications/graphql/__generated__/LightApplicationsOfProduct';
import { QueryBilling_billingQuery } from '../../dashboard/molecules/common/qraphql/__generated__/QueryBilling';
import {
    TeamByProductId_teamByProductId,
    TeamByProductId_teamByProductId_teamMembers
} from '../../teams/molecules/TeamMembers/graphql/__generated__/TeamByProductId';
import {
    QueryCurrentUser_currentUser,
    QueryCurrentUser_currentUser_roles
} from '../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { SelectItemProps } from '../molecules/SearchableSelect/SearchableSelectPresenter';

export const mockGraphQLError: GraphQLError = {
    message: 'FATALITY',
    name: 'error',
    locations: undefined,
    path: undefined,
    nodes: undefined,
    positions: undefined,
    source: undefined,
    originalError: undefined,
    extensions: undefined
};

export const mockSelectedDomain: QueryAllDomains_allDomains = {
    id: 102,
    name: 'Engineering Platform',
    __typename: 'Domain'
};

export const mockSelectedProduct: QueryAllProducts_allProducts = {
    id: 50051,
    logo: null,
    name: 'a',
    __typename: 'LightProductDTO'
};

export const mockedApplications: LightApplicationsOfProduct_lightApplicationsOfProduct[] = [
    {
        id: 50051,
        staticPictureId: null,
        name: 'a',
        service: false,
        __typename: 'LightApplicationDTO'
    }
];

export const mockFunction = jest.fn();

export const mockDataEmpty: QueryBilling_billingQuery[] = [];

// tslint:disable-next-line:no-any
export const createMockedState = (loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const userProductMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const preSelectedProduct: QueryAllProducts_allProducts = {
    __typename: 'LightProductDTO',
    id: 'ALL',
    logo: 'ALL',
    name: 'All Products'
};

export const currentUserMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: preSelectedProduct
        }
    }
});

export const selectedProductMockedState = (
    loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>
): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const buildDataExpectedEmptyResults = {
    labels: [],
    datasets: [
        {
            data: []
        }
    ]
};

export const mockErrorMessage = 'Error message';

export const mockTeamMember: TeamByProductId_teamByProductId_teamMembers = {
    __typename: 'User',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    lastName: 'Test',
    id: '11',
    roles: [
        {
            name: 'WRITE_ALL',
            __typename: 'Role'
        },
        {
            name: 'a-admin',
            __typename: 'Role'
        }
    ]
};

export const mockTeamByProductId: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: 2355,
    productOwner: {
        id: 11,
        firstName: 'TestUser',
        lastName: 'Test',
        __typename: 'User'
    },
    teamMembers: [
        {
            __typename: 'User',
            email: 'test1@mediamarktsaturn.com',
            firstName: 'Test',
            id: '1234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                }
            ]
        },
        {
            __typename: 'User',
            email: 'test2@mediamarktsaturn.com',
            firstName: 'Test2',
            id: '2234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                },
                {
                    name: 'test-admin',
                    __typename: 'Role'
                }
            ]
        }
    ]
};

export const mockCurrentUser: QueryCurrentUser_currentUser = {
    __typename: 'UserDTO',
    activated: true,
    createdBy: 'anonymousUser',
    createdDate: '2019-05-27T10:53:29.944312Z',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    id: 49050,
    imageContent: null,
    langKey: null,
    lastModifiedBy: 'Test',
    lastModifiedDate: '2019-06-28T05:56:51.175813Z',
    lastName: 'Test',
    login: 'Test',
    roles: [
        {
            concretePermissions: [],
            id: 1300,
            name: 'DP God',
            readOnly: true,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 4651,
            name: 'READ_ALL',
            readOnly: false,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 69002,
            name: 'a-read',
            readOnly: true,
            __typename: 'Role'
        }
    ],
    staticPictureId: null,
    subscribedToNewsletter: true
};

export const mockEmptyTeam: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: null,
    productOwner: null,
    teamMembers: []
};

export const mockCurrenUserRole: QueryCurrentUser_currentUser_roles = {
    concretePermissions: [],
    id: 1300,
    name: 'DP God',
    readOnly: true,
    __typename: 'Role'
};

export const mockForm = {
    getFieldDecorator: () => () => false,
    getFieldError: mockFunction,
    getFieldsError: mockFunction,
    getFieldsValue: mockFunction,
    getFieldValue: mockFunction,
    isFieldValidating: mockFunction,
    setFields: mockFunction,
    setFieldsValue: mockFunction,
    validateFields: mockFunction,
    validateFieldsAndScroll: mockFunction,
    isFieldsTouched: mockFunction,
    isFieldTouched: mockFunction,
    resetFields: mockFunction
};

export const mockHistory = {
    length: 7,
    push: mockFunction,
    action: 'PUSH',
    location: {
        hash: '',
        pathname: '/path',
        search: '',
        state: {}
    },
    replace: mockFunction,
    go: mockFunction,
    goBack: mockFunction,
    goForward: mockFunction,
    block: mockFunction,
    createHref: mockFunction,
    listen: mockFunction
};

export const mockSelectData: SelectItemProps[] = [
    {
        id: '351952',
        label: 'Test test'
    },
    {
        id: '80503',
        label: 'Test1 test1'
    },
    {
        id: '412605',
        label: 'Test2 test2'
    }
];
