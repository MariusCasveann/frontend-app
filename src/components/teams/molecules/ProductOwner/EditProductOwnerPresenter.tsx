import React from 'react';
import { Form, Modal } from 'antd';
import { GetFieldDecoratorOptions } from 'antd/lib/form/Form';
import { formLayout, getTeamFormRules } from '../../../createTeam/helpers/teamHelper';
import { TeamByProductId_teamByProductId } from '../TeamMembers/graphql/__generated__/TeamByProductId';
import SearchableSelectPresenter, {
    SelectItemProps
} from '../../../common/molecules/SearchableSelect/SearchableSelectPresenter';

interface EditProductOwnerPresenterProps {
    getFieldDecorator: (id: string, options: GetFieldDecoratorOptions) => (node: React.ReactNode) => React.ReactNode;
    team: TeamByProductId_teamByProductId;
    productOwnerOptions: SelectItemProps[];
    loadingProductOwner: boolean;
    setSearchProductOwnerInput: (value: string) => void;
    changeProductOwner: boolean;
    setChangeProductOwner: (flag: boolean) => void;
    handleSubmit: (e: React.MouseEvent<HTMLElement>) => void;
}

export default (props: EditProductOwnerPresenterProps) => {
    const {
        getFieldDecorator,
        team,
        changeProductOwner,
        setChangeProductOwner,
        handleSubmit,
        productOwnerOptions,
        loadingProductOwner,
        setSearchProductOwnerInput
    } = props;
    const rules = getTeamFormRules();

    if (changeProductOwner) {
        return (
            <Modal
                title="Edit product owner"
                visible={changeProductOwner}
                onCancel={() => setChangeProductOwner(false)}
                onOk={handleSubmit}
            >
                <Form layout="vertical" className="team-form">
                    <Form.Item label="Product owner" {...formLayout}>
                        {getFieldDecorator('productOwner', {
                            rules: rules.productOwner,
                            initialValue: (team && team.productOwner && team.productOwner.id) || undefined
                        })(
                            <SearchableSelectPresenter
                                data={productOwnerOptions}
                                mode="single"
                                loading={loadingProductOwner}
                                onSearchCallback={setSearchProductOwnerInput}
                            />
                        )}
                    </Form.Item>
                </Form>
            </Modal>
        );
    }

    return null;
};
