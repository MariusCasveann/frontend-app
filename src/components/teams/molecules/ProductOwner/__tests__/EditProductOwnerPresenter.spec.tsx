import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

jest.doMock('antd', () => ({ Form: 'Form', 'Form.Item': 'Form.Item' }));
jest.doMock('../../../../common/molecules/SearchableSelect/SearchableSelectPresenter', () => ({
    SearchableSelectPresenter: 'SearchableSelectPresenter'
}));

// tslint:disable-next-line:no-any
const _ShallowRenderer: any = ShallowRenderer;
const renderer = new _ShallowRenderer();

import EditProductOwnerPresenter from '../EditProductOwnerPresenter';
import { mockForm, mockFunction, mockSelectData, mockTeamByProductId } from '../../../../common/mock/MockData';

describe('EditProductOwnerPresenter', () => {
    it('should return the expected output when changeProductOwner is true ', () => {
        const tree = renderer.render(
            <EditProductOwnerPresenter
                getFieldDecorator={mockForm.getFieldDecorator}
                team={mockTeamByProductId}
                changeProductOwner={true}
                setChangeProductOwner={mockFunction}
                handleSubmit={mockFunction}
                productOwnerOptions={mockSelectData}
                loadingProductOwner={false}
                setSearchProductOwnerInput={mockFunction}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should return the expected output when changeProductOwner is false ', () => {
        const tree = renderer.render(
            <EditProductOwnerPresenter
                getFieldDecorator={mockForm.getFieldDecorator}
                team={mockTeamByProductId}
                changeProductOwner={false}
                setChangeProductOwner={mockFunction}
                handleSubmit={mockFunction}
                productOwnerOptions={mockSelectData}
                loadingProductOwner={false}
                setSearchProductOwnerInput={mockFunction}
            />
        );
        expect(tree).toMatchSnapshot();
    });
});
