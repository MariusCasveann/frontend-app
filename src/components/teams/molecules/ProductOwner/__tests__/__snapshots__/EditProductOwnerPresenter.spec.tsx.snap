// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProductOwnerPresenter should return the expected output when changeProductOwner is false  1`] = `null`;

exports[`EditProductOwnerPresenter should return the expected output when changeProductOwner is true  1`] = `
<Modal
  confirmLoading={false}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  onOk={[MockFunction]}
  title="Edit product owner"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form
    className="team-form"
    colon={true}
    hideRequiredMark={false}
    layout="vertical"
    onSubmit={[Function]}
  >
    <FormItem
      hasFeedback={false}
      label="Product owner"
      labelCol={
        Object {
          "span": 14,
        }
      }
      wrapperCol={
        Object {
          "span": 14,
        }
      }
    />
  </Form>
</Modal>
`;
